wallet.go

```diff
@@ -1122,6 +1124,102 @@ func (w *Wallet) DumpWIFPrivateKey(addr btcutil.Address) (string, error) {
 	return wif.String(), nil
 }
 
+// ImportAddress imports a watch-only address to the wallet.
+func (w *Wallet) ImportAddress(addr btcutil.Address, bs *waddrmgr.BlockStamp,
+	rescan bool) (string, error) {
+
+	// The starting block for the key is the genesis block unless otherwise
+	// specified.
+	if bs == nil {
+		bs = &waddrmgr.BlockStamp{
+			Hash:   *activeNet.Params.GenesisHash,
+			Height: 0,
+		}
+	}
+
+	// Check whether addr is P2SH or P2PKH and import accordingly
+	_, netID, err := base58.CheckDecode(addr.String())
+	if err != nil {
+		return "", err
+	}
+	switch {
+	case chaincfg.IsPubKeyHashAddrID(netID):
+		_, err := w.Manager.ImportAddress(addr, bs)
+		if err != nil {
+			return "", err
+		}
+	case chaincfg.IsScriptHashAddrID(netID):
+		_, err := w.Manager.ImportScript(addr.ScriptAddress(), bs)
```

Comment:
I'd also like to note that the tests must not be thorough enough to not catch this.